lib/rjson/handler.rb

```diff
@@ -30,7 +32,17 @@ def push o
 
     def result
       root = @stack.first.last
-      process root.first, root.drop(1)
+      output = process root.first, root.drop(1)
+      if @truncated
+        if output.is_a? Hash
+          output["_truncated"] = true 
+        elsif output.is_a? Array
+          output.push "_truncated"
+        else
+          throw "JSON doc is a single, truncated scalar"
```

Comment:
This is an unconventional use of `throw` (or at least I'm not clear on its purpose). What's the intention here?